### PR TITLE
Tarball creation in circle-ci

### PR DESCRIPTION
**Pull request by ferrandinand**
Creates a tarball local package

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -26,7 +26,64 @@ e2eTest: &e2eTest
...
+                  curl --user-agent curl-ci-sync -sSL -o "${HELM_TARBALL}" "${HELM_URL}/${HELM_TARBALL}"
+                  tar xzfv "${HELM_TARBALL}"
+                  PATH="$(pwd)/linux-amd64/:$PATH"
+                  helm init --client-only
+                }
+
+                echo "Setting up Helm package..."
+                setup_helm_client
+
+                sed -i -e "s/\[\[ \.SHA \]\]/${CIRCLE_SHA1}/g" $(find $HELM_DIR -type f)
```
Comment (ferrandinand) [inline .circleci/config.yml]: @kopiczko How do you feel about this?

File: .circleci/config.yml [hunk 2]
```diff
@@ -26,7 +26,64 @@ e2eTest: &e2eTest
...
+          - run:
+              name: "Creates a tarball file"
+              command: |
+                set -o errexit
+                set -o nounset
+                set -o pipefail
+      
+                readonly HELM_DIR=helm/${CIRCLE_PROJECT_REPONAME}-chart
+                readonly HELM_URL=https://storage.googleapis.com/kubernetes-helm
+                readonly HELM_TARBALL=helm-v2.11.0-linux-amd64.tar.gz
```
Comment (kopiczko) [inline .circleci/config.yml]: I'd prefer to:

* Have variables defined below `echo_*` functions as this part is just copy/paste and I'd like to skip that while reading scripts.
* Have scripts variables in lower case. This is to distinguish them from environment variables. This is bash good practice.

File: .circleci/config.yml [hunk 3]
```diff
@@ -26,7 +26,64 @@ e2eTest: &e2eTest
...
+      
+                echo_green() {
+                  echo -e "\e[32m${1}"
```
Comment (kopiczko) [inline .circleci/config.yml]: I updated those:

```suggestion
                  echo -e "\e[32m${1}\e[0m"
```

File: .circleci/config.yml [hunk 4]
```diff
@@ -26,7 +26,64 @@ e2eTest: &e2eTest
...
+                }
+                echo_red() {
+                  echo -e "\e[31m${1}"
```
Comment (kopiczko) [inline .circleci/config.yml]:
```suggestion
                  echo -e "\e[31m${1}\e[0m"
```

File: .circleci/config.yml [hunk 5]
```diff
@@ -26,7 +26,64 @@ e2eTest: &e2eTest
...
+                }
+                echo_yellow() {
+                  echo -e "\e[33m${1}"
```
Comment (kopiczko) [inline .circleci/config.yml]:
```suggestion
                  echo -e "\e[33m${1}\e[0m"
```

File: .circleci/config.yml [hunk 6]
```diff
@@ -26,7 +26,64 @@ e2eTest: &e2eTest
...
+
+                setup_helm_client() {
+                  echo "Setting up Helm client..."
+                  curl --user-agent curl-ci-sync -sSL -o "${HELM_TARBALL}" "${HELM_URL}/${HELM_TARBALL}"
+                  tar xzfv "${HELM_TARBALL}"
+                  PATH="$(pwd)/linux-amd64/:$PATH"
+                  helm init --client-only
+                }
+
+                echo "Setting up Helm package..."
```
Comment (kopiczko) [inline .circleci/config.yml]:
```suggestion
```

File: .circleci/config.yml [hunk 7]
```diff
@@ -26,7 +26,64 @@ e2eTest: &e2eTest
...
+                }
+                echo_red() {
+                  echo -e "\e[31m${1}"
+                }
+                echo_yellow() {
+                  echo -e "\e[33m${1}"
+                }
+
+                setup_helm_client() {
+                  echo "Setting up Helm client..."
```
Comment (kopiczko) [inline .circleci/config.yml]:
```suggestion
                  echo "installing helm binary"
```

File: .circleci/config.yml [hunk 8]
```diff
@@ -26,7 +26,64 @@ e2eTest: &e2eTest
...
+                  curl --user-agent curl-ci-sync -sSL -o "${HELM_TARBALL}" "${HELM_URL}/${HELM_TARBALL}"
+                  tar xzfv "${HELM_TARBALL}"
+                  PATH="$(pwd)/linux-amd64/:$PATH"
+                  helm init --client-only
```
Comment (kopiczko) [inline .circleci/config.yml]:
```suggestion
                  helm init --client-only
                  echo "installed helm binary"
```

File: .circleci/config.yml [hunk 9]
```diff
@@ -26,7 +26,64 @@ e2eTest: &e2eTest
...
+
+                echo "Setting up Helm package..."
+                setup_helm_client
+
+                sed -i -e "s/\[\[ \.SHA \]\]/${CIRCLE_SHA1}/g" $(find $HELM_DIR -type f)
+
+                echo $(grep -nr $CIRCLE_SHA1 $HELM_DIR)
+                helm package ${HELM_DIR}
+
+                echo_green "ok: Created helm package"
```
Comment (kopiczko) [inline .circleci/config.yml]:
```suggestion
                echo_green "ok: created chart tarball"        
```

File: .circleci/config.yml (hunk 1 above)
Comment (kopiczko) [reply .circleci/config.yml]: I don't have an issue with the command itself, but we should replace `.SHA` with `.VERSION`.

Comment (ferrandinand) [reply .circleci/config.yml]: Using version makes me feel that we will need to change all charts and we can break some backward compability.
Changing SHA for version is just a matter of naming? Or are we replacing the current system x.x.x-SHA for only VERSION which contains only the SHA? With the second one we could lose the possibility of notifying major/minor version. Is this right?

Comment (kopiczko) [reply .circleci/config.yml]: We need to replace current `v1.0.0-[[ .SHA ]]` with `[[ .VERSION ]]`. Otherwise how we can set (git) tagged version?

Comment (ferrandinand) [reply .circleci/config.yml]: I understand now. Makes sense if we are doing that tagging strategy

File: .circleci/config.yml (hunk 2 above)
Comment (ferrandinand) [reply .circleci/config.yml]: I feel like they are constants more than scripts variables so it would make sense to keep them capitalized if so.

**Pull request closed without merging**